common/beerocks/tlvf/AutoGenerated/include/beerocks/tlvf/beerocks_message_common.h

```diff
@@ -1189,6 +1189,92 @@ typedef struct sSteeringEvSnr {
     }
 } __attribute__((packed)) sSteeringEvSnr;
 
+typedef struct sTriggerChannelScanParams {
+    sMacAddr radio_mac;
+    //time interval (msec) between channels during scan
+    uint32_t dwell_time_ms;
+    //size of provided channel_pool
+    uint8_t channel_pool_size;
+    //pool of channels to be scaned
+    uint8_t channel_pool[beerocks::message::SUPPORTED_CHANNELS_LENGTH];
+    void struct_swap(){
+        radio_mac.struct_swap();
+        tlvf_swap(32, reinterpret_cast<uint8_t*>(&dwell_time_ms));
+    }
+    void struct_init(){
+        radio_mac.struct_init();
+        dwell_time_ms = 0x0;
+        channel_pool_size = 0x0;
+    }
+} __attribute__((packed)) sTriggerChannelScanParams;
+
+typedef struct sChannelScanRequestParams {
+    //an invalid (-1) value indicates this value is not requested
+    int32_t dwell_time_ms;
+    //an invalid (-1) value indicates this value is not requested
+    int32_t interval_time_sec;
+    //an invalid (-1) value indicates this value is not requested
+    int8_t channel_pool_size;
+    uint8_t channel_pool[beerocks::message::SUPPORTED_CHANNELS_LENGTH];
+    void struct_swap(){
+        tlvf_swap(32, reinterpret_cast<uint8_t*>(&dwell_time_ms));
+        tlvf_swap(32, reinterpret_cast<uint8_t*>(&interval_time_sec));
+    }
+    void struct_init(){
+        dwell_time_ms = -0x1;
+        interval_time_sec = -0x1;
+        channel_pool_size = 0x0;
+    }
+} __attribute__((packed)) sChannelScanRequestParams;
+
+typedef struct sChannelScanResults {
```

Comment:
I think we talked about this one - it should be refactored to not using strings, so it should first be done here.

Reply:
refactored